### PR TITLE
add tool for evaluating cel expresssions

### PR DESCRIPTION
**Pull request by ljupchokotev**
# Changes

Adding a tool that can be used to evaluate cel expressions locally. Useful if you have a complex cel expression that you need to debug. Referencing secrets is not supported, I am not sure how I would go about implementing that.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Add `cel-eval` tool for evaluating cel expressions from HTTP requests.
```

### CODE REVIEW

**Comment by ljupchokotev** [comment]
Looks like the failed tests are not related to any of my code changes?

File: cmd/cel-eval/cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+	mapStrDyn := decls.NewMapType(decls.String, decls.Dyn)
+	env, err := cel.NewEnv(
+		triggerscel.Triggers("default", secretLister{}),
+		celext.Strings(),
+		celext.Encoders(),
+		cel.Declarations(
+			decls.NewVar("body", mapStrDyn),
+			decls.NewVar("header", mapStrDyn),
+			decls.NewVar("extensions", mapStrDyn),
```
Comment (dibyom) [inline cmd/cel-eval/cmd/root.go:91]: would be nice to add a way for users to specify `extensions` via the command line as well.

File: cmd/cel-eval/cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+func (sl secretLister) Secrets(namespace string) corev1lister.SecretNamespaceLister {
+	return secretLister{}
+}
+
+func (sl secretLister) Get(name string) (*v1.Secret, error) {
+	return nil, nil
+}
+
+func evalCEL(w io.Writer, expressionPath, httpPath string) error {
+	// Read HTTP request.
```
Comment (dibyom) [inline cmd/cel-eval/cmd/root.go:66]: wrong comment :)

File: cmd/cel-eval/cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	expression, err := readExpression(expressionPath)
```
Comment (dibyom) [inline cmd/cel-eval/cmd/root.go:67]: for CEL expressions it would be nice to read them from the CLI directly as well.

**Review by dibyom — APPROVED** [review]
Thanks @ljupchokotev This will be very useful

Could you add a Readme to the cmd/cel-eval folder with an example usage?

One minor doc comment and a couple of nice to have features that can be added later!

File: cmd/cel-eval/cmd/root.go (hunk 1 above)
Comment (ljupchokotev) [reply cmd/cel-eval/cmd/root.go:91]: something like `--extension key=value`?

File: cmd/cel-eval/cmd/root.go (hunk 3 above)
Comment (ljupchokotev) [reply cmd/cel-eval/cmd/root.go:67]: ``--expression '<expression>'`?

File: cmd/cel-eval/cmd/root.go (hunk 1 above)
Comment (dibyom) [reply cmd/cel-eval/cmd/root.go:91]: yup! (and later maybe `--extension-file extensions.json` or something like that)

File: cmd/cel-eval/cmd/root.go (hunk 3 above)
Comment (dibyom) [reply cmd/cel-eval/cmd/root.go:67]: yup!

**Comment by dibyom** [comment]
@ljupchokotev Sorry it took me so long to get to this! Thank you so much. This will be very useful!

/lgtm

**Pull request merged**